projects/packages/my-jetpack/_inc/state/reducers.js

```diff
@@ -6,10 +6,54 @@ import { combineReducers } from '@wordpress/data';
 /**
  * Internal dependencies
  */
-import { SET_PURCHASES, SET_PURCHASES_IS_FETCHING } from './actions';
+import {
+	SET_PURCHASES,
+	SET_PURCHASES_IS_FETCHING,
+	SET_PRODUCT_ACTION_ERROR,
+	SET_PRODUCT_STATUS,
+	IS_FETCHING_PRODUCT_STATUS,
+} from './actions';
 
-const products = ( state = {} ) => {
-	return state;
+const products = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case SET_PRODUCT_STATUS: {
+			const { productId, status } = action;
+			return {
+				...state,
```

Comment:
Looking at the code, looks like we have nothing else in the `products` state than `items`, maybe isn't the case to return the items directly?

```javascript
return {
  items: {
     ...
  }
  error: {},
}
```

Reply:
I'm not sure... I think in general we should consider keep the prev state of the state 😅 even when, for now, it has items and error keys

Reply:
> Looking at the code, looks like we have nothing else in the products state than items,...

I think it can change in a short time. We need to decide how we'd like to structure the data in general. For instance, we need to handle errors at the products level. If so, it will bring changes here.
I think we could store these errors in a new `errors` array, where each field is a product slug.

```js
{
  items: {
    backup: {
      description: ...
    },
  },
  errors: {
    backup: {
      code: 'not_found',
      // ...
  },
},
```

Does te make sense?

Reply:
@retrofox yeah, makes sense, it remembers a lot of `formik` approaches of handling data and errors.